CMakeLists.txt

```diff
@@ -0,0 +1,120 @@
+#
+# CMakeLists file for OpenSSL
+# Copyright 2022-2023 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the Apache License 2.0 (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+#
+
+cmake_minimum_required(VERSION 3.17)
+
+project(OpenSSL C CXX)
+
+# Configure the OpenSSL project
+if(OPENSSL_CONFIGURE_OPTIONS)
+  #set(ENV{BUILDFILE_DEBUG_RULES} 1)
+  #set(ENV{BUILDFILE_DEBUG_DEPENDS} 1)
+  if (NOT PERL_EXECUTABLE)
+    find_package(Perl REQUIRED)
```

Comment:
I would use here
`find_package(PerlLibs 5.10.0 REQUIRED)`
You stated Perl minimal version 5.10.0 in the latest 3.0.7 release.
Moreover, I think openssl uses some modules of Perl? (Correct me if I am wrong)
[FindPerlLibs](https://cmake.org/cmake/help/latest/module/FindPerlLibs.html) searches for perl executable and runtime libs.

Reply:
We try to limit the perl module use to the modules that are included in the perl distribution as of 5.10.  There's one distinct exception: Text::Template, which we have solved by including it into our source distribution (with the blessing of the author)

Reply:
That's great that you're reducing dependencies!
Then please add version number to existing find_package code.